Model/Sinch.php

```diff
@@ -2052,6 +2059,51 @@ private function addCategoryData(
                 ON DUPLICATE KEY UPDATE
                     value = c.Description"
             );
+
+            $this->_doQuery(
+                "INSERT INTO $catalog_category_entity_text
+                (
+                    attribute_id,
+                    store_id,
+                    entity_id,
+                    value
+                )
+                (
+                    SELECT
+                    $this->_categoryMetadescriptionAttrId,
+                    0,
+                    scm.shop_entity_id,
+                    c.MetaDescription
+                    FROM $categories_temp c
+                    JOIN $sinch_categories_mapping scm
+                        ON c.store_category_id = scm.store_category_id
+                )
+                ON DUPLICATE KEY UPDATE
+                    value = c.MetaDescription"
+            );
+
+            $this->_doQuery(
+                "INSERT INTO $catalog_category_entity_text
+                (
+                    attribute_id,
+                    store_id,
+                    entity_id,
+                    value
+                )
+                (
+                    SELECT
+                    $this->_categoryDescriptionAttrId,
```

Comment:
You're adding a query to populate catalog_category_entity_text with the data for category description (which is correct), but you're leaving the pointless insert into catalog_category_entity_varchar for the same attribute about 50 lines above this. Will not merge in current state (bugfixes are welcome, but this introduces data duplication). Please make sure to check the backend type on attributes you insert data for, and ONLY insert data into 1 EAV table